File: spieleplattform-backend/src/main/kotlin/com/example/spieleplattformbackend/models/Game.kt
package com.example.spieleplattformbackend.models

import com.fasterxml.jackson.annotation.JsonManagedReference
import jakarta.persistence.*
import java.time.LocalDate

@Entity
class Game(
    @Column(nullable = false)
    var name: String,
    @Column(nullable = true)
    var releaseDate: LocalDate,
    @Column(nullable = true)
    var developer: String,
    @OneToMany(mappedBy = "game", fetch = FetchType.EAGER)
    @JsonManagedReference
    var ratings: List<Rating> = emptyList(),
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int? = null
) {
}

File: spieleplattform-frontend/src/components/Game/Game.ts
import Rating from "../Rating/Rating"
import GameConsole from "../GameConsole/GameConsole";

export default interface Game {
    id: number
    name: string
    releaseDate: Date
    developer: string
    author: string
    description: string
    imageUrl: string
    youtubeId: string
    gameConsoles: Array<GameConsole>
    ratings: Array<Rating>
}
